src/ring_theory/non_zero_divisors.lean

```diff
@@ -32,7 +32,45 @@ def non_zero_divisors (R : Type*) [monoid_with_zero R] : submonoid R :=
 
 localized "notation R`⁰`:9000 := non_zero_divisors R" in non_zero_divisors
 
-variables {M M' M₁ : Type*} [monoid_with_zero M] [monoid_with_zero M'] [comm_monoid_with_zero M₁]
+variables {M M' M₁ R R' : Type*} [monoid_with_zero M] [monoid_with_zero M']
+  [comm_monoid_with_zero M₁] [ring R] [comm_ring R']
+
+lemma mem_non_zero_divisors_iff {r : M} : r ∈ M⁰ ↔ ∀ x, x * r = 0 → x = 0 := iff.rfl
+
+@[simp] lemma mul_right_mem_non_zero_divisors_eq_zero_iff {x r : M} (hr : r ∈ M⁰) :
```

Comment:
This simp-lemma will match pretty often, I presume. But I don't think `r ∈ M⁰` will often be provable by `simp`. So I'm inclined to leave it out of the default simp-set.

Idem dito for the other lemmas below.